File: Tutorial/Examples/Topic_routing/Consumer_Python/Consumer_2.py
import pika
from pika.exchange_type import ExchangeType


def on_message_received(channe, method, properties, body):
    print(f'[Area consumer] Received message: {body}')


# Set connection parameters. If connecting to a real server localhost should be replaced by the server´s address.
connection_parameters = pika.ConnectionParameters('localhost')

# Instantiate a connection unsing the connection_parameters previously defined
connection = pika.BlockingConnection(connection_parameters)

# Instantiate a channel
channel = connection.channel()

# Instantiate an exchange and queue.
channel.exchange_declare(exchange='topic', exchange_type=ExchangeType.topic)
queue = channel.queue_declare(queue='', exclusive=True)

# Bind the queue using the "square" and "both" keys.
channel.queue_bind(exchange='topic',
                   queue=queue.method.queue, routing_key='square.area')


# Start consuming messages
channel.basic_consume(queue=queue.method.queue, auto_ack=True,
                      on_message_callback=on_message_received)

print('[Area consumer] Started consuming messages')

channel.start_consuming()

File: Tutorial/Examples/Topic_routing/Consumer_C#/Consumer_2/Consumer_2.cs
using System;
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;

class Program{
    static void Main(string[] args){
        // Create connection.  When connecting to a real server, localhost should be replaced by the server´s address.
        var factory = new ConnectionFactory{ HostName = "localhost"};

        // Instantiate the connection and one channel.
        using var connection = factory.CreateConnection();
        using var channel = connection.CreateModel();

        // Declare an exchange called topic.
        channel.ExchangeDeclare(exchange: "topic", type: ExchangeType.Topic);

        // Declare a queue and bind it using the "square.area" key.
        var queueName = channel.QueueDeclare().QueueName;
        channel.QueueBind(queue: queueName, exchange: "topic", routingKey: "square.area");

        // Declare consumer.
        var consumer = new EventingBasicConsumer(channel);

        // Subscribe to the Received event.
        consumer.Received += (model, ea) =>{
            // Get the body of the message as byte array, decode to string and print it.
            var body = ea.Body.ToArray();
            var message = Encoding.UTF8.GetString(body);
            Console.WriteLine($"[Area consumer] Message received: {message}");
        };

        // Start consuming messages.
        channel.BasicConsume(queue: queueName, autoAck: true, consumer: consumer);
        Console.WriteLine("[Area consumer] Started consuming");

        // Wait for a keypress to exit.
        Console.ReadKey();
    }
}